docs/google/devices.md

```diff
@@ -0,0 +1,194 @@
+
+Qubits on Google devices are laid out in a grid structure.  Connectivity is limited to
+adjacent qubits, either horizontally or vertically.  
+
+Measurement takes much longer than other gates.  Currently, the only supported
+configuration is to have terminal measurement in the final moment of a circuit.
+
+Most devices have a limited number of gates that can applied to the device.  Gates not
+in that set must be decomposed into an equivalent circuit using gates within the set.
+See below for those restrictions. 
+
+## Best Practices
+
+This section lists some best practices for creating a circuit that performs well on hardware
+devices.  This is an area of active research, so users are encouraged to try multiple
+approaches to improve results.
+
+### Use built-in optimizers as a first pass
+
+Using built-in optimizers will allow you to compile to the correct .  As they are
+automated solutions, they will not always perform as well as a hand-crafted solution, but
+it provides a good starting point for creating a circuit that is likely to run successfully
+on hardware.
+
+```python
+import cirq
+import cirq.google as cg
+
+
+# Create your circuit here
+my_circuit = cirq.Circuit()
+
+# Convert the circuit onto a Google device.
+# Specifying a device will verify that the circuit satisifies constraints of the device
+# Specifying an optimizer type (e.g. 'sqrt_iswap' or 'sycamore') will convert to the appropriate
+# gate set and also do basic optimization, such as combining successive one-qubit gates. 
+sycamore_circuit = cg.optimized_for_sycamore(my_circuit, new_device=cg.Sycamore, optimizer_type='sqrt_iswap')
+```
+
+### Good moment structure
+
+In an attempt to fulfill the user's desire, quantum devices will respect the moment
+structure of circuits to the greatest extent possible.
+
+To this end, it is important that the moment structure of the circuit is kept as
+short and concise as possible.  The length of a moment will generally be the length
+of the longest gate in the moment, so keeping gates with similar durations together
+will generally benefit the performance.
+
+In particular, keep measurement gates in the same moment and make sure that any
+circuit optimizers run do not alter this by pushing measurements forward.  This
+behavior can be avoided by measuring all qubits with a single gate or by adding
+the measurement gate after all optimizers have run.
+
+### Short gate depth
+
+In the current NISQ (noisy intermediate scale quantum) era, gates and devices still
+have significant error.  Both gate errors and T1 decay rate can cause long circuits
+to have noise that overwhelms any signal in the circuit.  Also, circuits that are
+much too long can even be rejected due to various limits in the system.
+
+The recommended gate depths vary significantly with the structure of the circuit itself
+and will likely increase as the devices improve.  However, circuits with more than
+50 moments will generally struggle to retain any information.
```

Comment:
is this true?

Reply:
(I don't think it is)

Performance generally depends on width * depth

> A rough guide can be found by multiplying together gate fidelities to get a sense for the probability of success for an algorithm. For example, if you have 20 moments on 20 qubits, you can roughly estimate circuit fidelity as (0.995)^(20*20)

Reply:
Thanks.  This is much more precise.  I'll link to calibration metrics once we release them.